network-api/networkapi/mozfest/models.py

```diff
@@ -59,6 +59,61 @@ def clean(self):
             raise exceptions.ValidationError("Please provide both link URL and link text, or neither")
 
 
+@register_snippet
+class NewsletterSignupWithBackground(TranslatableMixin):
```

Comment:
This looks very similar to the `CTA` classes from the `campaigns` module. 

Most of the fields and even the core markup seem to be shared with the general `Signup` and `NewsletterSignupBlock` and the same JS seems to be responsible for the actual functionality. 

Would it make sense to reuse some of those definitions?

Reply:
I'm reluctant too, but yes I'll inherit the CTA. I'm of the opinion there is _far_ too much inheritance in this codebase, but I take your point given the are both technically sign ups :)

Reply:
I tend to agree with that assessment of too much inheritance. In this case though, the behavior is coupled and needs to match so that they can be handled by the same JS. 

Maybe the split of classes is at a weird place. E.g. instead of having a bunch of different CTA snippet classes (that are each probably only used once), why not make the styling and option on the instance 🤷‍♂️

If you see a good reason to not use inheritance here, then please feel free stick with the separate definition.